### DISCUSSION TITLE
Table Filters for bulk action (Office hours: Oct 20, 2020)

### DISCUSSION THREAD

**Discussion by ishah-twilio** [twilio-labs/paste#865] [Office hours] [answered]
Hello,

I'm working on Twilio IoT Super SIM Bulk action features which requires filtering of thousands of data records. 
 I had two design concepts for data table filters. I had tested two options with 7 users. 

Option1 - existing UI pattern — where filter/search is part of the table header
Option2 - was based on messaging insights filter feature  — where filter is separately called out and part of the disclosure component which can be expanded on a need base.

 2nd concept was well received by users. I like DSys guidance, feedback and review on the proposed design.
 
Questions I like guidance or help with:

1. Is it ok to use disclosure component? 
2. Feedback/Guidance on usage of clearing filters
3. Feedback/Guidance on summary of filters
4. Guidance on combining Radio / textarea
5. Grid / spacing and layout of filters 

Concept design mocks is added here and detailed walk through will be added in the the office hour discussion signup sheet. 

![Dsys - github discussion](https://user-images.githubusercontent.com/69016180/96321022-f0a2a980-0fc8-11eb-8e86-1fef92e0239b.png)

**Comment by SiTaggart** [comment]
Hi @ishah-twilio, someone from the Design team will likely get back to you next week.

I do have some clarifying questions to get us started though.

1. Are the number of filters static? Is it only those 4 fields you can filter by? Is the max number of filters 4?

2. What is the initial state of the disclosure? Is it collapsed or expanded? Is that state of disclosure persistent upon the customer returning to the page or does it revert back to what ever the initial state was?

3. In a similar vein are the filters persistent? If I create a filter on this page and come back to it another day, is the same filter applied, or does it always start off in an unfiltered state?

4. Can you walk me through what lead you to providing a customer the ability to remove a filter from a collapsed state?

## Textarea and Radio group

The requirement for this is that the textarea really needs to have a visible label in this layout. 

Every form element needs to have its own label, you can't reuse another inputs label as there is no technical way to associate a label to more than one input.

In its current form, what makes this hard is it's not visually obvious which is the label for the textarea, so in that case, you must provide one.

If the radio group was stack vertically, you could argue that by visual association you could _visually_ use Unique Name as the label for a Textarea underneath it (and hide the extra one). It still goes against our guidelines though, we do ask you to provide a visible label for every form field.

**Reply by ishah-twilio** [reply]
Answers to your questions:

1. Are the number of filters static? Is it only those 4 fields you can filter by? Is the max number of filters 4?

> Yes for phase 1. In future, there could be more filters. For example, SIM Connectivity Status (online / offline). Though that data is not available as of today.

2. What is the initial state of the disclosure? Is it collapsed or expanded? Is that state of disclosure persistent upon the customer returning to the page or does it revert back to what ever the initial state was?

> Initial state: Minimized
> Should be persistent till user is in same session. Though I need to discuss this with engineering team. I will bring eng team to the meeting. As of now, I don't have any user feedback for this usecase.
> 

In a similar vein are the filters persistent? If I create a filter on this page and come back to it another day, is the same filter applied, or does it always start off in an unfiltered state?

> This filters are only per browser session. If user. logout out or come back any other day it will go back to unfiltered state.

Can you walk me through what lead you to providing a customer the ability to remove a filter from a collapsed state?

> SSIM users changes page size from 10 to 100 and like to have focus on the data/table. Though in the usability study, none of the users minimized the filter.

**Comment by SiTaggart** [comment, marked as answer]
So this is an interesting insight from testing

> Though in the usability study, none of the users minimized the filter

Coupled with the fact that the initial state is minimized but not persistent beyond session, and filters are not persistent beyond session, I'm not entirely sure showing applied filters in the disclosure trigger is worth while.

If I return the page and there are no filters already applied, and I have to open it anyway to set the filters, and user are not likely to collapse it again, I feel like you should just show the selected filters inside the disclosure body.
